fiftyone/zoo/datasets/__init__.py

```diff
@@ -176,8 +182,111 @@ def download_zoo_dataset(
+            loader. Only required if different from the repo's name.
+            This will be the name used to register the dataset in the zoo.
+            You can change this by setting the `zoo_name` arg.
+            Must implement 'prepare_and_download'
+        splits (None): a list of splits to download, if applicable. Typical
+            values are ``("train", "validation", "test")``. If None,
+            all available splits are downloaded. Consult the documentation
+            for the :class:`ZooDataset` you specified to see the supported splits
+        dataset_dir (None): the directory into which to download the dataset.
+            By default, it is downloaded to a subdirectory of
+            ``fiftyone.config.dataset_zoo_dir``
+        overwrite (False): whether to overwrite any existing files
+        cleanup (True): whether to cleanup any temporary files generated during
+            download
+
+    Returns:
+        tuple of
+
+        -   info: the :class:`ZooDatasetInfo` for the dataset
+        -   dataset_dir: the directory containing the dataset
+    """
+
+    # https://gist.github.com/kaixi-wang/ed4eb4f4dd29e944a1a5de762a9511a0#file-coco-2017-py
+    # https://github.com/kaixi-wang/dummy_dataset/tree/main
+    # https://github.com/kaixi-wang/dummy_dataset/blob/main/dummy_dataset.py
+    # kaixi-wang/dummy_dataset
+
+    def _parse_url(url):
+        # params = re.search('github.com/(?P<user>[A-Za-z0-9_-]+)/.+#file-(?P<dataset>[A-Za-z0-9_-]+).py', url).groupdict()
+        print(url)
+        params = re.search(
+            "github.com/(?P<user>[A-Za-z0-9_-]+)/(((?P<repo>["
+            "A-Za-z0-9_-]+)((/.+?)?/(?P<branch>[A-Za-z0-9_-]+)?(.py)?)?)|("
+            "?P<gist>.+#file-(?P<dataset>[A-Za-z0-9_-]+).py))",
+            str(url),
+        ).groupdict()
+        params["input_type"] = "url"
+        return params
+
+    def _parse_repo(repo):
+        params = repo.split("/")
+        return {
+            "input_type": "repo",
+            "user": params[0],
+            "dataset": params[1],
+            "branch": params[2] if len(params) > 2 else "main",
+        }
+
+    if etaw.is_url(repo_or_url):
+        src_dict = _parse_url(repo_or_url)
+    else:
+        src_dict = _parse_repo(repo_or_url)
+    if not src_dict.get("dataset"):
+        src_dict["dataset"] = src_dict.get("repo")
+    dataset_name = src_dict.get("dataset") or src_dict.get("repo")
+    user = src_dict.get("user")
+    if not dataset_dir:
+        dataset_dir = fo.config.dataset_zoo_dir
+    # zoo_dataset, dataset_dir = _parse_dataset_details(
+    #     dataset_name, dataset_dir
+    # )
+    # split = splits[0] if len(splits) == 1 else None
+    if not name:
+        name = dataset_name
+    # dataset_module = httpimport.load(name,
+    #                               repo_or_url)
+    print("user", user, "dataset_name", dataset_name)
+    with httpimport.github_repo(user, dataset_name, ref="main"):
```

Comment:
I think this should be implemented more simply:
1. Download the github repo to disk 
2. Import the module that contains the `download_and_prepare()` function, then execute it

Prior art
- https://pytorch.org/docs/stable/hub.html#loading-models-from-hub
- https://pytorch.org/docs/stable/_modules/torch/hub.html#load

Our spec proposes the standard of having a `.fiftyone` file in the repository that, in turn, contains any metadata about the zoo dataset and could contain the name of the entrypoint`.py` file that contains the `download_and_prepare.py` function.

Reply:
It's important that the github repository be explicitly downloaded in step 1, because that's what plugins expect: the source code living in some directory on disk.

Reply:
this file was an old file I just referenced to copy some of the url parsing. please disregard